### PR TITLE
Zanderz/core 7128/restore console

### PR DESCRIPTION
**Pull request by zanderz**
At some point we became more aggressive about doing os.Exit() with an error code on a failed CLI command, which may be reasonable but messed up some Defer calls.

Probably the more correct long term solution is to wrap a different "main outer" with a single os.Exit() at the end.

@maxtaco @taruti

### CODE REVIEW

**Comment by maxtaco** [comment]
delegating my review to @patrickxb

File: go/keybase/main.go [hunk 1]
```diff
@@ -89,6 +89,7 @@ func main() {
 		}
 	}
 	if g.ExitCode != keybase1.ExitCode_OK {
+		logger.RestoreConsoleMode()
 		os.Exit(int(g.ExitCode))
```
Comment (taruti) [inline go/keybase/main.go]: Perhaps write a helper function?
```
func keybaseExit(exitCode int) {
    logger.RestoreConsoleMode()
 	os.Exit(exitCode)
}
```

And use that in all three places?

**Comment by patrickxb** [comment]
yup

File: go/keybase/main.go [hunk 2]
```diff
@@ -40,6 +40,11 @@ func handleQuickVersion() bool {
 	return false
 }
 
+func keybaseExit(exitCode int) {
+    logger.RestoreConsoleMode()
```
Comment (maxtaco) [inline go/keybase/main.go]: it looks like the spacing is off here. did it pass `go fmt`?

**Comment by maxtaco** [comment]
ah, the spacing is breaking your CI run.   you should install the pre-commit hooks

**Comment by zanderz** [comment]
I never got them (pre-commit hooks) working on windows, sorry

**Pull request merged**